woocommerce/woocommerce-android
# [Woo POS] M2: Products loading errors and empty state

Closes: #12001


### Description

* Adds empty and error state to the products list
* Hides cart for this states
* I do not pass errors from the splash screen as this seems to increase complexity for very small UX improvement (we may consider that to improve later)
* I changed the API of the ProductsSource class as I could not figure out how to properly implement the separation between full loading state and loading with pull to refresh (@samiuelson I think your advice will be really welcome here)

### Testing information

[Fixed_displayong_of_the_empty_state.patch](https://github.com/user-attachments/files/16360207/Fixed_displayong_of_the_empty_state.patch)
* Use the patch or your own imitations of the empty list and error
* Notice the empty and error states
* Overal products related (pagination, reloading) should be retested

### Images/gif



https://github.com/user-attachments/assets/4d1653d4-5db4-4e49-a3f4-577c67ffc868
